From b478d4ef6c669a73c7b54cab1e8b07a813aa1377 Mon Sep 17 00:00:00 2001
From: Dongjin Kim <tobetter@gmail.com>
Date: Mon, 11 May 2020 07:41:52 +0900
Subject: [PATCH 067/109] ODROID-C4: arm64/dts: add more CPU frequencies

Two new CPU frequencies, 2.016GHz and 2.100GHz are added.
ODROID-C4 could run 2.016GHz, but 2.100GHz can cause unexpected crashing.

Change-Id: I87588077509bcb1165b59ad94ff06f682a193c84
Signed-off-by: Dongjin Kim <tobetter@gmail.com>
---
 arch/arm64/boot/dts/amlogic/meson64_odroidc4.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson64_odroidc4.dts b/arch/arm64/boot/dts/amlogic/meson64_odroidc4.dts
index 9b2c191620b9..b7e7df08691e 100644
--- a/arch/arm64/boot/dts/amlogic/meson64_odroidc4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson64_odroidc4.dts
@@ -10,3 +10,18 @@
 / {
 	model = "Hardkernel ODROID-C4";
 };
+
+&cpu_opp_table {
+	opp-2016000000 {
+		opp-hz = /bits/ 64 <2016000000>;
+		opp-microvolt = <1000000>;
+	};
+	opp-2100000000 {
+		opp-hz = /bits/ 64 <2100000000>;
+		opp-microvolt = <1022000>;
+	};
+};
+
+&vddcpu {
+	regulator-max-microvolt = <1030000>;
+};
-- 
2.25.1

